src/methods/pseudoperiodic.jl

```diff
@@ -40,40 +41,39 @@ function surrogenerator(x::AbstractVector, pp::PseudoPeriodic, rng = Random.defa
     Ñ = length(z)
     w = zeros(eltype(z), Ñ-1) # weights vector
     y = Dataset([z[1] for i in 1:N])
+    s = similar(x)
     init = (y = y, w = w, z = z)
-    return SurrogateGenerator(pp, x, init, rng)
+    return SurrogateGenerator(pp, x, s, init, rng)
 end
 
 function (sg::SurrogateGenerator{<:PseudoPeriodic})()
     y, z, w = getfield.(Ref(sg.init), (:y, :z, :w))
     ρ, shift = getfield.(Ref(sg.method), (:ρ, :shift))
-    pseudoperiodic!(y, sg.x, z, w, ρ, shift, sg.rng)
+    pseudoperiodic!(sg.s, y, sg.x, z, w, ρ, shift, sg.rng)
 end
 # Low-level method, also used in `noiseradius`
-function pseudoperiodic!(y, x, z, w, ρ, shift, rng)
+function pseudoperiodic!(s, y, x, z, w, ρ, shift, rng)
     N, Ñ = length.((x, z))
     y[1] = shift ? rand(rng, z.data) : z[1]
     @inbounds for i in 1:N-1
         w .= (exp(-norm(z[t] - y[i])/ρ) for t in 1:Ñ-1)
         j = sample(rng, 1:Ñ-1, pweights(w))
         y[i+1] = z[j+1]
     end
-    return y[:, 1]
+    s .= y[:, 1]

```

Comment:
```suggestion
    for (i, p) in enumerate(y); s[i] = p[1]; end
```
This eliminates creating an intermediate vector via `y[:, 1]`.